channeldb/payments.go

```diff
@@ -94,10 +94,17 @@ const (
 	// destination was found during path finding.
 	FailureReasonNoRoute FailureReason = 1
 
+	// FailureReasonError indicates that an unexpected error happened during
+	// payment.
+	FailureReasonError FailureReason = 2
+
+	// FailureReasonIncorrectPaymentDetails indicates that either the hash
+	// is unknown or the final cltv delta or amount is incorrect.
+	FailureReasonIncorrectPaymentDetails FailureReason = 3
```

Comment:
not a blocker (related to my other comment tho), but are these serialized anywhere? if not, it might be useful to define 0 as the unknown case so that forgetting to set it results in an unknown error rather than the an incorrect-but-known error

Reply:
We started serializing this to the db with the reliable payments PRs. It cannot be changed without a db migration.

That was the advantage of the pointer approach. You were forced to handle it.